python/lsst/pipe/tasks/dcrAssembleCoadd.py

```diff
@@ -702,5 +737,29 @@ def fillCoadd(self, dcrModels, skyInfo, tempExpRefList, weightList, calibration=
             self.assembleMetadata(coaddExposure, tempExpRefList, weightList)
             coaddUtils.setCoaddEdgeBits(model[skyInfo.bbox].mask, model[skyInfo.bbox].variance)
             coaddExposure.setMaskedImage(model[skyInfo.bbox])
+            if mask is not None:
+                coaddExposure.setMask(mask)
             dcrCoadds.append(coaddExposure)
         return dcrCoadds
+
+    def calculateGain(self, modelIter):
+        """Calculate the gain to use for the current iteration.
+
+        Parameters
+        ----------
+        modelIter : `int`
+            The current iteration of forward modeling.
+
+        Returns
+        -------
+        gain : `float`
+            Relative weight to give the new solution when updating the model.
+        """
+        if self.config.useProgressiveGain:
```

Comment:
the particular values of these numbers do not seem well motivated, can you add a comment explaining why these magic numbers and not others? Or include a link to a paper or something where they come from